### ISSUE TITLE
[Bug]: GPU plugin can't finish correctly

### ISSUE THREAD

**Issue by Wq-dd** [openvinotoolkit/openvino#25141]
### OpenVINO Version

2024.2.0

### Operating System

Windows System

### Device used for inference

GPU

### Framework

PyTorch

### Model used

llama2 7b 

### Issue description

When I using openvino==2024.2.0 to run qwen2-7b-int4 and llama2-7b-int4 on iGPU, I got an inference error 
```
"Due to the driver bug any subsequent OpenCL API call will cause application hang, so GPU plugin can't finish correctly. Please try to update the driver or reduce memory consumption (use smaller batch size, less streams, lower precision, etc)to avoid CL_OUT_OF_RESOURCES exception"
```
Then I change the device to CPU, and there is no error.
Is there a bug of openvino or somothing I did wrong? Please help.

### Step-by-step reproduction

1. llama2 7b int4 model quantized by optimum-cli
2. intel ultra 155h PC
3. openvino==2024.2.0

### Relevant log output

```shell
"Due to the driver bug any subsequent OpenCL API call will cause application hang, so GPU plugin can't finish correctly. Please try to update the driver or reduce memory consumption (use smaller batch size, less streams, lower precision, etc)to avoid CL_OUT_OF_RESOURCES exception"
```

### Issue submission checklist

- [X] I'm reporting an issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [X] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by andrei-kochin** [comment]
@Wq-dd have you tried to update the driver like the error suggested?

**Comment by Wq-dd** [comment]
Yes, I have updated the driver to the latest driver version==31.0.101.5590, but I still got the same error .

**Comment by isanghao** [comment]
How much memory does your system have?

**Comment by Wq-dd** [comment]
16GB

**Comment by chux0519** [comment]
How large is your model? As far as I know, there is an issue with oneDNN / SYSCL where there is a bug when allocating memory exceeding 4GB on GPUs.

check: https://github.com/oneapi-src/oneDNN/issues/1638

**Comment by p-durandin** [comment]
Please check OV 2025.2 and modern GPU driver (newer than 32.0.101.6452)
